crazystocks/ARCHITECTURE.md

```diff
@@ -0,0 +1,105 @@
+# Goals
+
+The goal is to produce a small toolkit and library for evaluating stock market strategies based on predictive approaches.
+
+By predictive approaches, we mean machine learning on quantities related to the stock market. The quantities may include prices, volumes, expected gains/losses under some action.
+
+By stragegies, we mean simple rule based strategies which use the predictions of the machine learning models
+and may apply simple techniques as thresholding, stop-loss rules, time of day and so on.
+
+The same code should be able to run on:
+
+- the analyst's laptop using the DBG public dataset
+- the quantopian.com website
+
+# Stretch goals
+
+Ideally, we should be able to demonstrate consistent gains on the models/strategies across multiple stocks
+and also on quantopian.com. The strategies should be robust to slippage and costs.
+
+# Non-goals
+
+We don't aim to apply the most sophisticated modeling approaches (e.g. LSTM, multi-headed Attention Mechanism, very deep neural networks) at the intial stage. We use two basic models Linear Regression and Boosted Decision Trees together
+with informative features.
+
+Our focus is on robustness and usability of code. We want to enable ease of experimentation and ease of validation on quantopian.com. 
+
+# Toolkit
+
+![Toolkit Architecture](tools.png)
+
+The toolkit consists of the following commands
+
+## Data Downloader
```

Comment:
Your best bet for this is to create a secondary layer above one of the listed backtesters. This layer is a data processor - call it `dbg_data_kit` - that downloads and transforms DBG data into something a more robust and popular backtester can use.

Why reinvent the wheel?

Reply:
Will check against the frameworks you mentioned. Perhaps it will be better to do it this way.